### PR TITLE
[sw] Cleanup how Register Headers are used

### PR DESCRIPTION
**Pull request by lenary**
Before this change, a build directory would contain multiple copies of
each of the generated register definition headers. We now almost
exclusively use these headers from inside the DIF implementations (or
the device drivers, for those hardware ip blocks without DIFs yet), and
do not consider the definitions to be a public interface.

With this in mind, taking any `<ip>_regs.h` file out of the meson-
declared sources for a dependency means it will not be rebuilt by any
library that depends on the implementation that uses the header.

This cuts down on warnings from generating the headers, and ensures
that only one copy of each header should exist in the build directory.

The exception here is `usbdev_regs.h` which is currently used by
`hello_usbdev.c`, but this should be sorted when a USB device DIF is
completed.

---

There is also a commit that does some minimal cleanups to the meson
target definitions: ensuring that all the `*_regs.h` have the same target
suffix, and that DIF target definitions match each other.

### CODE REVIEW

**Comment by gkelly** [comment]
@mcy can you take a look for the Meson things?

File: sw/device/examples/hello_usbdev/meson.build [hunk 1]
```diff
@@ -5,7 +5,9 @@
 foreach device_name, device_lib : sw_lib_arch_core_devices
   hello_usbdev_elf = executable(
     'hello_usbdev_' + device_name,
-    sources: ['hello_usbdev.c'],
+    sources: [
```
Comment (silvestrst) [inline sw/device/examples/hello_usbdev/meson.build:8]: Why this change? Isn't it supposed to be on a single line when there is only a single element inside the brackets?

File: sw/device/lib/dif/meson.build [hunk 2]
```diff
@@ -23,10 +23,12 @@ dif_plic = declare_dependency(
   link_with: static_library(
     'dif_plic_ot',
     sources: [
-      hw_top_earlgrey_rv_plic_h,
+      hw_top_earlgrey_rv_plic_reg_h,
       'dif_plic.c',
     ],
-    dependencies: [sw_lib_mmio]
```
Comment (silvestrst) [inline sw/device/lib/dif/meson.build:29]: Same as above.

File: sw/device/examples/hello_usbdev/meson.build (hunk 1 above)
Comment (lenary) [reply sw/device/examples/hello_usbdev/meson.build:8]: Doing it this way makes it a lot easier to add other sources or dependencies (future changes will have fewer diff lines). I'm not sure we have a meson style guide at the moment.

Comment (imphil) [reply sw/device/examples/hello_usbdev/meson.build:8]: I guess we don't need a style guide for everything. Keeping things on separate lines and with a trailing comma is good practice to keep diffs small and reduce the probability of merge conflicts (independent of the language).

Comment (mcy) [reply sw/device/examples/hello_usbdev/meson.build:8]: Not a fan of the new style, but I think Sam's justification is pretty solid, as far as git greping for stuff goes (I don't think it'll be as helpful with merge conflicts but w/e).
